spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestBranchDDL.java

```diff
@@ -206,9 +212,54 @@ public void testCreateBranchUseCustomMaxRefAge() throws NoSuchTableException {
                 tableName, branchName, maxRefAge));
   }
 
-  private Table createDefaultTableAndInsert2Row() throws NoSuchTableException {
-    sql("CREATE TABLE %s (id INT, data STRING) USING iceberg", tableName);
+  @Test
+  public void testDropBranch() throws NoSuchTableException {
+    insertRows();
+
+    Table table = validationCatalog.loadTable(tableIdent);
+    String branchName = "b1";
+    table.manageSnapshots().createBranch(branchName, table.currentSnapshot().snapshotId()).commit();
+    SnapshotRef ref = table.refs().get(branchName);
+    Assert.assertEquals(table.currentSnapshot().snapshotId(), ref.snapshotId());
+
+    sql("ALTER TABLE %s DROP BRANCH %s", tableName, branchName);
+    table.refresh();
+
+    ref = table.refs().get(branchName);
+    Assert.assertNull(ref);
+  }
+
+  @Test
+  public void testDropBranchDoesNotExist() {
+    AssertHelpers.assertThrows(
+        "Cannot perform drop branch on branch which does not exist",
+        IllegalArgumentException.class,
+        "Branch does not exist: someBranch",
+        () -> sql("ALTER TABLE %s DROP BRANCH %s", tableName, "someBranch"));
```

Comment:
[nit] liked the branch name of 'nonExistingBranch'  used with IF EXISTS clause, can we use it here as well ?